src/services/jwt.service.ts

```diff
@@ -0,0 +1,33 @@
+import jwt from 'jsonwebtoken';
+import { User } from '@entities/user.entity';
+import { JWT_SECRET, ACCESS_TOKEN_LIFE } from '@config';
+
+export const createJWT = (user: User): Promise<string> => {
+  return new Promise((resolve, reject) => {
+    try {
+      const token = jwt.sign(
+        { data: { userId: user.id, email: user.email } },
+        JWT_SECRET || '',
+        { expiresIn: ACCESS_TOKEN_LIFE || 21600 }
+      );
+      resolve(token);
+    } catch (error) {
+      console.log('decode error: ', error);
+      reject(new Error('Error creating JWT'));
+    }
+  });
+};
+
+export const decodeJWT = (
+  token: string
+): Promise<string | { [key: string]: any } | null> => {
+  return new Promise((resolve, reject) => {
+    try {
+      const decoded = jwt.decode(token);
```

Comment:
Is it ok to use decode here, or should it be [verify](https://www.npmjs.com/package/jsonwebtoken#jwtdecodetoken--options)?

Reply:
The intention here was to decode the token so it's ok as it is. Maybe the question would be "was it ok to decode the token inside the authentication middleware or maybe it was just fine with verify?". I'm not sure but I got the impression that both should work, with a small difference on performance maybe. I'll take a look later

Reply:
Decode: returns the payload without verifying.
Verify: returns the payload if verifications pass (signature, expired, etc.)
👍 👏